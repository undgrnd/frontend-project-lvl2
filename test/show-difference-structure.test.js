const { getDifference } = require('../src/get-difference');
const { getStructureReportDifference } = require('../src/format/get-structure-report-difference');

const before = require('./__fixtures__/before-nesting.json');
const after = require('./__fixtures__/after-nesting.json');

test('Show difference as JSON', () => {
  const diff = getDifference(before, after);

  expect(getStructureReportDifference(diff)).toMatchSnapshot();
});
