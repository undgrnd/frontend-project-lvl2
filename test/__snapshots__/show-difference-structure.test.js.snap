// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show difference as JSON 1`] = `"[{\\"action\\":\\"object modified\\",\\"name\\":\\"common\\",\\"value\\":[{\\"action\\":\\"added\\",\\"name\\":\\"follow\\",\\"value\\":false},{\\"action\\":\\"not modified\\",\\"name\\":\\"setting1\\",\\"value\\":\\"Value 1\\"},{\\"action\\":\\"deleted\\",\\"name\\":\\"setting2\\",\\"value\\":200},{\\"action\\":\\"deleted while modifying\\",\\"name\\":\\"setting3\\",\\"value\\":true},{\\"action\\":\\"added while modifying\\",\\"name\\":\\"setting3\\",\\"value\\":{\\"key\\":\\"value\\"}},{\\"action\\":\\"modified\\",\\"name\\":\\"setting3\\",\\"oldValue\\":true,\\"value\\":{\\"key\\":\\"value\\"}},{\\"action\\":\\"added\\",\\"name\\":\\"setting4\\",\\"value\\":\\"blah blah\\"},{\\"action\\":\\"added\\",\\"name\\":\\"setting5\\",\\"value\\":{\\"key5\\":\\"value5\\"}},{\\"action\\":\\"object modified\\",\\"name\\":\\"setting6\\",\\"value\\":[{\\"action\\":\\"not modified\\",\\"name\\":\\"key\\",\\"value\\":\\"value\\"},{\\"action\\":\\"added\\",\\"name\\":\\"ops\\",\\"value\\":\\"vops\\"}]}]},{\\"action\\":\\"object modified\\",\\"name\\":\\"group1\\",\\"value\\":[{\\"action\\":\\"deleted while modifying\\",\\"name\\":\\"baz\\",\\"value\\":\\"bas\\"},{\\"action\\":\\"added while modifying\\",\\"name\\":\\"baz\\",\\"value\\":\\"bars\\"},{\\"action\\":\\"modified\\",\\"name\\":\\"baz\\",\\"oldValue\\":\\"bas\\",\\"value\\":\\"bars\\"},{\\"action\\":\\"not modified\\",\\"name\\":\\"foo\\",\\"value\\":\\"bar\\"},{\\"action\\":\\"deleted while modifying\\",\\"name\\":\\"nest\\",\\"value\\":{\\"key\\":\\"value\\"}},{\\"action\\":\\"added while modifying\\",\\"name\\":\\"nest\\",\\"value\\":\\"str\\"},{\\"action\\":\\"modified\\",\\"name\\":\\"nest\\",\\"oldValue\\":{\\"key\\":\\"value\\"},\\"value\\":\\"str\\"}]},{\\"action\\":\\"deleted\\",\\"name\\":\\"group2\\",\\"value\\":{\\"abc\\":12345}},{\\"action\\":\\"added\\",\\"name\\":\\"group3\\",\\"value\\":{\\"fee\\":100500}}]"`;
